### PR TITLE
A whole bunch of python packages necessary to install ROS

### PR DESCRIPTION
**Pull request by expipiplus1**
Note that I had to disable the tests for `rosdep` as they depended on network access.
Someone interested in Python may want to take a look to see if these individual tests can be disabled or fixed.
###### Things done
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FRidh** [comment]
please add yourself as maintainer of these packages.

**Comment by zimbatm** [comment]
Looking good. Can you use the new pypi mirror for the sources ? There are examples available in the same file.

**Comment by expipiplus1** [comment]
I'm not going to go any further with it; ROS turned out to be a complete quagmire of despair.

**Comment by zimbatm** [comment]
Okay. Was there other issues than the pypi mirrors ?

**Comment by expipiplus1** [comment]
I didn't try the pypi mirrors.

I went into this not realizing the true nature of ROS. It's not wrong when it
bills itself as an "operating system". 

As far as I can tell, the ROS engineers didn't really know how to use CMake but
they did know how to use Python; the consequence of this being that although
the packages ostensibly build with CMake they're actually completely dependent
on the "catkin" packaging system. As I'm sure you know, packaging systems are
hard to get right; catkin is as cumbersome as you would expect a mish-mash of
CMake and Python to be. I found no utility in ROS which couldn't exist as a
stand alone binary or library. There is no need for such a Byzantine packaging
system.

The default install of ROS contains such gems as `roscp` and `rosls` which wrap
`cp` and `ls` but do strange and different things in the presence of a file
named `package.xml`; this is the calibre of design one deals with when one uses
ROS.

I managed to extract the data I needed out of a "ROS bag" by installing ROS on
a VM. I do not need to use ROS beyond that on this project. Even installing it
on a VM was a chore. They've managed to make ROS so environment dependent that
although it installed on Ubuntu Trusty it fails to build on Utopic or Vivid.

tl;dr: ROS is a convoluted pile of junk that I pray I've seen the last of.

**Comment by zimbatm** [comment]
Ouch. Thanks for describing the issue on great length, hopefully it will serve as a warning for future packagers :)
